### PR TITLE
Run tests on PRs.

### PR DESCRIPTION
**Pull request by AdamHillier**

### CODE REVIEW

**Comment by AdamHillier** [comment]
Ah maybe this is not what we want after all, I thought from the documentation that if two things triggered an event then it would only run once, but clearly not...

**Comment by AdamHillier** [comment]
The solution may be to run tests on PR and then protect master with the rule that says that tests have to have been run on a rebased version

**Comment by lgeiger** [comment]
> The solution may be to run tests on PR and then protect master with the rule that says that tests have to have been run on a rebased version

I think we can also limit branch tests to only run on master using [something like this](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#onpushpull_requestbranchestags)

**Comment by AdamHillier** [comment]
I think this might finally be working

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -1,6 +1,10 @@
 name: Lint
 
-on: [push]
+on:
+  push:
+    branches:
+      - master
+  pull_request: {}
```
Comment (lgeiger) [inline .github/workflows/lint.yml]: Would this work too?
```suggestion
  pull_request
```

Comment (AdamHillier) [reply .github/workflows/lint.yml]: I tried that but apparently not :( ebe00a8d61f4c380a4996f80b5cc996d141d29f0

**Pull request merged**